packages/editor-ui/src/stores/workflows.store.ts

```diff
@@ -1465,6 +1465,43 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, () => {
 		return !!matchedChatNode;
 	}
 
+	//
+	// Start Canvas V2 Functions
+	//
+
+	function removeNodeById(nodeId: string): void {
+		const node = getNodeById(nodeId);
+		if (!node) {
+			return;
+		}
+
+		removeNode(node);
```

Comment:
I wonder if we need to pass around the full `NodeUI`, both for `removeNode` and `removeAllNodeConnection`. It seems like we only care about `nodeName`, similar to `clearNodeExecutionData`

Reply:
We only need the `node.name`. I'm reusing the existing function here though. There's an ongoing task for this: https://linear.app/n8n/issue/N8N-7342/migrate-workflows-store-to-retrieve-all-workflow-data-by-workflow-id